RELEASE/scripts/autoscend/auto_edTheUndying.ash

```diff
@@ -692,76 +636,46 @@ boolean ed_eatStuff()
-	}
-	if((item_amount($item[Xiblaxian Ultraburrito]) > 0) && (my_fullness() == 0) && (fullness_limit() >= 4) && (item_amount($item[Astral Hot Dog]) == 0))
-	{
-		autoEat(1, $item[Xiblaxian Ultraburrito]);
-	}
-	if((my_level() >= 11) && ((my_fullness() + 3) <= fullness_limit()) && (item_amount($item[Astral Hot Dog]) > 0))
-	{
-		autoEat(1, $item[Astral Hot Dog]);
-	}
-	if((my_level() >= 9) && ((my_fullness() + 3) <= fullness_limit()) && (item_amount($item[Astral Hot Dog]) > 0) && (my_adventures() < 4))
-	{
-		autoEat(1, $item[Astral Hot Dog]);
-	}
-	if(!get_property("_fancyHotDogEaten").to_boolean() && (my_daycount() == 1) && (my_level() >= 9) && ((my_fullness() + 3) <= fullness_limit()) && (item_amount($item[Astral Hot Dog]) == 0) && (my_adventures() < 10) && (item_amount($item[Clan VIP Lounge Key]) > 0))
-	{
-		eatFancyDog("video games hot dog");
-	}
-	if(get_property("auto_dickstab").to_boolean() && !get_property("_fancyHotDogEaten").to_boolean() && (my_daycount() == 1) && ((my_fullness() + 2) <= fullness_limit()) && (item_amount($item[Astral Hot Dog]) == 0) && (item_amount($item[Clan VIP Lounge Key]) > 0) && chateaumantegna_available())
-	{
-		eatFancyDog("sleeping dog");
-	}
-	if((my_daycount() >= 3) && (my_inebriety() == 0) && (inebriety_limit() == 4) && (item_amount($item[Xiblaxian Space-Whiskey]) > 0) && (my_adventures() < 10))
-	{
-		autoDrink(1, $item[Xiblaxian Space-Whiskey]);
-	}
-	if((item_amount($item[Astral Pilsner]) > 0) && ((my_inebriety() + 1) <= inebriety_limit()) && (my_level() >= 11))
-	{
-		autoDrink(1, $item[Astral Pilsner]);
-	}
-	if((item_amount($item[Astral Pilsner]) > 0) && ((my_inebriety() + 1) <= inebriety_limit()) && (my_level() >= 10) && (my_adventures() < 3))
-	{
-		autoDrink(1, $item[Astral Pilsner]);
-	}
-	if((item_amount($item[Astral Pilsner]) > 0) && ((my_inebriety() + 1) <= inebriety_limit()) && (my_level() >= 9) && (my_adventures() < 3) && (my_fullness() >= fullness_limit()))
-	{
-		autoDrink(1, $item[Astral Pilsner]);
-	}
-	if((item_amount($item[Coinspiracy]) >= 6) && ((my_inebriety() + 3) <= inebriety_limit()) && (my_adventures() < 3) && (item_amount($item[Astral Pilsner]) == 0))
-	{
-		buyUpTo(1, $item[Highest Bitter]);
-		autoDrink(1, $item[Highest Bitter]);
-	}
+	// ideally, we should only need the above in this function as the code below 
+	// should be handled by consumeStuff();
 
+	// expose semi-rare counters
 	if (!contains_text(get_counters("Fortune Cookie", 0, 200), "Fortune Cookie"))
 	{
-		if((item_amount($item[Clan VIP Lounge Key]) > 0) && (my_meat() >= 500) && (inebriety_limit() == 4) && ((my_inebriety() == 0) || (my_inebriety() == 3)) && (auto_get_clan_lounge() contains $item[Clan Speakeasy]))
+		boolean shouldEatCookie = (my_meat() >= npc_price($item[Fortune Cookie]) && fullness_left() > 0 && my_level() < 12);
+		if (inebriety_left() > 0)
 		{
-			autoDrink(1, $item[Lucky Lindy]);
+			shouldEatCookie = (shouldEatCookie && !autoDrink(1, $item[Lucky Lindy]));
```

Comment:
consider pill keeper?

Reply:
I would but my test account doesn't have any IotMs.
Also if we want to be "optimal" about using the SRs for the pill keeper, Ed is the worst class to use it past the free use per day as his turngen comes from his spleen (he has to buy his stomach and liver and only gets 5 in both) and we would want to expose the counter, use the counter SR, use the pill keeper SR (1 or more times) then expose the counter again etc.

Reply:
Yeah, we have fortune cookie, lucky lindy use sprinkled all over the place. I figured this like the others probably deserve a more focused refactoring to make better use of SRs